snappy/click.go

```diff
@@ -481,6 +482,47 @@ func removePackageBinaries(m *snapYaml, baseDir string) error {
 	return nil
 }
 
+func addPackageDesktopFiles(m *snapYaml, baseDir string) error {
+	if err := os.MkdirAll(dirs.SnapDesktopFilesDir, 0755); err != nil {
+		return err
+	}
+
+	desktopFiles, err := filepath.Glob(filepath.Join(baseDir, "meta", "gui", "*.desktop"))
+	if err != nil {
+		return fmt.Errorf("cannot get desktop files for %v: %s", baseDir, err)
+	}
+
+	for _, df := range desktopFiles {
+		content, err := ioutil.ReadFile(df)
+		if err != nil {
+			return err
+		}
+
+		realBaseDir := stripGlobalRootDir(baseDir)
+		content = bytes.Replace(content, []byte("${SNAP}"), []byte(realBaseDir), -1)
+
```

Comment:
I don't think it's a good idea to use a desktop file verbatim. It needs to be associated with a command and have the Exec line match that command. For instance my desktop Exec could read data at the permissions level of the shell before going into confinement. I think that a much better approach is to generate the desktop file from data in the YAML.